data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/PipelineParser.java

```diff
@@ -110,13 +110,18 @@ private void buildPipelineFromConfiguration(
                     .map(this::newProcessor)
                     .collect(Collectors.toList());
 
-            final List<List<Processor>> decoratedProcessorSets = processorSets.stream()
-                    .map(processorSet -> processorSet.stream()
+            final Map<List<Processor>, String> processorMap = pipelineConfiguration.getProcessorPluginSettings().stream()
```

Comment:
The order here is very important.

I'd much rather have a wrapper around `Processor` rather than trying to force a `Map` here.

Reply:
Like have a wrapper which returns processor name? This again leads to having plugin id's, which makes this logic simple here.

Reply:
Something like this:


```
 private static class IdentifiedComponent<T> {
        private final T component;
        private final String name;

        private IdentifiedComponent(final T component, final String name) {
            this.component = component;
            this.name = name;
        }

        T getComponent() {
            return component;
        } 
        
        String getName() {
            return name;
        }
    }

    private List<IdentifiedComponent<Processor>> newProcessor(final PluginSetting pluginSetting) {
        final List<Processor> processors = pluginFactory.loadPlugins(
                Processor.class,
                pluginSetting,
                actualClass -> actualClass.isAnnotationPresent(SingleThread.class) ?
                        pluginSetting.getNumberOfProcessWorkers() :
                        1);
        
        return processors.stream()
                .map(p -> new IdentifiedComponent<>(p, pluginSetting.getName()))
                .collect(Collectors.toList());
    }
```

You will need to change how to handle the data coming back from `newProcessors`, but that shouldn't be too challenging.

Reply:
I made the changes but the static class will not be used once we have plugin ids.